### PR TITLE
Add Avalonia.ReactiveUI.Events based on Pharmacist

### PR DESCRIPTION
**Pull request by worldbeater**
## What does the pull request do?

This PR adds a new `Avalonia.ReactiveUI.Events` project which contains generated `.Events()` wrappers that turn all events in Avalonia assemblies into `IObservable<TEventArguments>` wrappers. [ReactiveUI.Events packages](https://www.reactiveui.net/docs/handbook/events/) currently exist for nearly every platform supported by ReactiveUI, so what's stopping us from adding Avalonia support as well? ✨

Here is a GIF image that illustrates the power of the `*.Events` packages.

ezgif-6-61a74596899a

## What is the current behavior?

Currently, there is no `.Events()` wrappers for Avalonia.

## What is the updated/expected behavior with this PR?

Now, events are generated as `Events_Avalonia.cs` at Nuke Build time and placed into the `Avalonia.ReactiveUI.Events` dir.

## How was the solution implemented (if it's not obvious)?

We use [Pharmacist](https://github.com/reactiveui/pharmacist) to make this feature work. The `ReactiveUI.Events.XamForms` package is implemented as follows: https://github.com/reactiveui/ReactiveUI/blob/main/src/ReactiveUI.Events.XamForms/ReactiveUI.Events.XamForms.csproj#L13 In this PR, I used a similar implementation for the `Avalonia.ReactiveUI.Events` package. However, here we generate the events from compiled Avalonia assemblies and the `.Events()` generation process is a part of the NukeBuild pipeline.

This will require a new Avalonia NuGet package named `Avalonia.ReactiveUI.Events`. Might need help with configuring this.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -34,17 +34,10 @@ jobs:
   pool:
     vmImage: 'macOS-10.14'
   steps:
-  - task: UseDotNet@2
-    displayName: 'Use .NET Core SDK 3.1.101'
+  - task: DotNetCoreInstaller@0
```
Comment (grokys) [inline azure-pipelines.yml:37]: What's the difference between the `UseDotNet` and `DotNetCoreInstaller` tasks?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -34,17 +34,10 @@ jobs:
...
+    displayName: 'Use .NET Core SDK 3.1.302'
```
Comment (grokys) [inline azure-pipelines.yml:38]: We were previously stuck on 3.1.101 due to a [problem with sharpgentools](https://github.com/SharpGenTools/SharpGenTools/issues/164). Given that it looks like that must no longer be an issue (due to the build now passing) would it be worth upgrading to the latest 3.1 patch (3.1.401 it looks like?)

**Review by grokys — COMMENTED** [review]
Looks good to me, just got a few questions about the .NET core version upgrade.

File: azure-pipelines.yml (hunk 1 above)
Comment (worldbeater) [reply azure-pipelines.yml:37]: Well according to [DotNetCoreInstallerV0](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/DotNetCoreInstallerV0) and [UseDotNetV2](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/UseDotNetV2) readme files, the latter seems to be a newer version of the former. Planning to change the task back to `UseDotNet@2`, thanks for reviewing this!

File: azure-pipelines.yml (hunk 2 above)
Comment (worldbeater) [reply azure-pipelines.yml:38]: Seems like `3.1.401` is now deployed. Probably worth bumping the version now 🎉
https://github.com/actions/virtual-environments/issues/1407#issuecomment-681107768

**Comment by worldbeater** [comment]
Well with `3.1.401` the unit tests on macOS seem to hang forever. 
Probably worth retriggering the CI to verify, and if the issue persists, worth rolling back to `3.1.302`.
Now we need someone with admin permissions to retrigger the CI :P

File: azure-pipelines.yml (hunk 2 above)
Comment (worldbeater) [reply azure-pipelines.yml]: Done! Wondering if we need to adjust some additional stuff to ship `Avalonia.ReactiveUI.Events` to NuGet.

**Comment by worldbeater** [comment]
Feels like we can go with .NET Core `3.1.401`, no CI failures so far. Pinging @grokys with a hope for a review!

**Comment by grokys** [comment]
Yeah, CI seems to be stable so I'm 👍 . @kekekeks @danwalmsley for confirmation?

**Review by grokys — APPROVED** [review]
Ok, no votes against, so lets merge.

**Pull request merged**